core/keyserver/keyserver.go

```diff
@@ -258,9 +254,21 @@ func (s *Server) UpdateEntry(ctx context.Context, in *tpb.UpdateEntryRequest) (*
 		return nil, grpc.Errorf(codes.InvalidArgument, "Invalid mutation")
 	}
 
-	if err := s.queue.Enqueue(index[:], m); err != nil {
-		log.Printf("Enqueue error: %v", err)
-		return nil, grpc.Errorf(codes.Internal, "Write error")
+	// Save mutation to the database.
+	txn, err := s.factory.NewDBTxn(ctx)
+	if err != nil {
+		return nil, grpc.Errorf(codes.Internal, "Cannot create transaction")
+	}
+	if _, err := s.mutations.Write(txn, in.GetEntryUpdate().GetUpdate()); err != nil {
+		log.Printf("mutations.Write failed: %v", err)
+		if err := txn.Rollback(); err != nil {
+			log.Printf("Cannot rollback the transaction: %v", err)
```

Comment:
Previous error is already logged two lines above. The same behavior is used throughout keyserver.go